### ISSUE TITLE
The error of the training about arcface 

### ISSUE THREAD

**Issue by LONG520520** [cvqluu/Additive-Margin-Softmax-Loss-Pytorch#15]
hello：
When I was training arcface's second epoch, what was the reason for this?
![image](https://user-images.githubusercontent.com/86965285/165713881-4a7d6932-6a7d-477c-a780-52582954b95a.png)
This is the visual diagram of training:
![image](https://user-images.githubusercontent.com/86965285/165714306-2cdebe6a-2c46-4f16-80d2-5b63010bd624.png)

**Comment by msalman95** [comment]
I have the exact same issue, the Loss either goes to 0 or to NaN for arcface. Did you find the the solution for this?

**Comment by EduanKK** [comment]
> I have the exact same issue, the Loss either goes to 0 or to NaN for arcface. Did you find the the solution for this?

I have same issue. In all arcface, sphereface, cosface loss.

**Comment by alexkrz** [comment]
Same issue on my side for Arcface. Any updates on this one?

**Comment by alexkrz** [comment]
I have adjusted the loss function in my `dev` branch. Probably newer Pytorch versions introduced a change, such that the implementation in the `master` branch does not produce the expected outcomes anymore.
My branch is tested with Pytorch version 2.4.1.
You can find it here: https://github.com/alexkrz/arcmargin-losses/tree/dev
